src/App.test.js

```diff
@@ -1,8 +1,20 @@
 import { render, screen } from '@testing-library/react';
 import App from './App';
+import { MemoryRouter } from 'react-router-dom';
+import userEvent from '@testing-library/user-event';
 
-test('renders learn react link', () => {
-  render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+test('renders the character list', async () => {
+  render(
+    <MemoryRouter>
+      <App />
```

Comment:
You <App /> already has access to the routes so you shouldn't need a MemoryRouter here.